lib/blab.rb

```diff
@@ -0,0 +1,36 @@
+class Blab
+  attr_reader :id, :timestamp, :content, :user_id
+
+  def initialize(id, timestamp, content, user_id)
+    @id = id
+    @timestamp = timestamp
+    @content = content
+    @user_id = user_id
+  end
+
+  def self.all
+    result = DatabaseConnection.query("SELECT * FROM blabs;")
+    result.map do |blab|
+      Blab.new(blab['id'], blab['timestamp'], blab['content'], blab['user_id'])
+    end
+  end
+
+  def self.create(content, user_id)
+    result = DatabaseConnection.query("INSERT INTO blabs (content, user_id)" +
+      " VALUES ('#{content}', '#{user_id}')" +
+      " RETURNING id, timestamp, content, user_id;")
+    Blab.new(
+      result.first['id'],
```

Comment:
Efficiency nitpick - if you do:

```
blab_data = result.first

Blab.new(
  blab_data['id'], 
  blab_data['timestamp'] 
  ... etc 
)
```

It both reads better (as in, the 'result' is named more clearly) and is more efficient (because every time you call result.first you are doing a lookup, and even though there is probably only one result there is still a minor cost to this). The efficiency thing doesn't matter in this case, but it's still a good habit to be aware of that as there are cases when it does matter.